### ISSUE TITLE
dht.put succeeds if there are no closest peers

### ISSUE THREAD

**Issue by pgte** [libp2p/js-libp2p-kad-dht#32]
`ipfs.dht.put()` succeeds if there are no closest peers. Although it fails to save the record into any other peer, the call succeeds locally.

Shouldn't we be checking whether the peer set is empty [here](https://github.com/libp2p/js-libp2p-kad-dht/blob/master/src/index.js#L169-L172)?

**Comment by diasdavid** [comment]
There are always closest peers as long as there is one peer.

Can you write a test for the fail you are seeing?

**Comment by alanshaw** [comment]
Yes it's an edge case, but can happen when a node starts up with no connectivity and also in tests. We have the same issue with `provide`.

[`GetClosestPeers` in `go-libp2p-kad-dht` throws an error](https://github.com/libp2p/go-libp2p-kad-dht/blob/31765355df17ef818381169dc36180c84c119928/lookup.go#L57-L62).

IMHO that's not an error case, we should just check the array of returned peers when we call [`getClosestPeers`](https://github.com/libp2p/js-libp2p-kad-dht/blob/3439b074285ee7c0d985f2c92a6c39381322c4a9/src/index.js#L408) and throw an error in `put` and `provide` if it's empty.
